packages/typegpu/src/data/struct.ts

```diff
@@ -121,13 +127,16 @@ class TgpuStructImpl<TProps extends Record<string, AnyTgpuData>>
   }
 }
 
-function getAttribute(field: AnyTgpuData): string | undefined {
-  if (field instanceof TgpuAlignedImpl) {
+function getAttribute<T extends AnyTgpuData>(field: T): string | undefined {
```

Comment:
as noticed by @reczkok, this function doesn't work properly when both properties are present. we should fix that in another PR